src/test/java/com/commercetools/sync/producttypes/utils/producttypeactionutils/BuildAttributeDefinitionUpdateActionsTest.java

```diff
@@ -881,6 +926,135 @@ void buildAttributesUpdateActions_WithSetOfIdenticalLEnums_ShouldBuildNoActions(
+        // preparation
+        final AttributeDefinitionDraft newDefinition = AttributeDefinitionDraftBuilder
+            .of(StringAttributeType.of(), "a", ofEnglish("new_label"), true)
+            .build();
+        final ProductTypeDraft productTypeDraft = ProductTypeDraftBuilder
+            .of("foo", "name", "desc", singletonList(newDefinition))
+            .build();
+
+        final LocalizedEnumAttributeType oldLenumType =
+            LocalizedEnumAttributeType.of(singletonList(LocalizedEnumValue.of("foo", ofEnglish("bar"))));
+
+        final AttributeDefinition oldDefinition = AttributeDefinitionBuilder
+            .of("a", ofEnglish("new_label"), oldLenumType)
+            .build();
+        final ProductType productType = mock(ProductType.class);
+        when(productType.getAttributes()).thenReturn(singletonList(oldDefinition));
+
+        // test
+        final List<UpdateAction<ProductType>> updateActions = buildAttributesUpdateActions(productType,
+            productTypeDraft, SYNC_OPTIONS);
+
+        // assertions
+        assertThat(updateActions).containsExactly(
+            RemoveAttributeDefinition.of("a"),
+            AddAttributeDefinition.of(AttributeDefinitionDraftBuilder
+                .of(newDefinition.getAttributeType(),
+                    newDefinition.getName(), newDefinition.getLabel(),
+                    newDefinition.isRequired())
+                .build())
+        );
+    }
+
+    @Test
+    void buildAttributesUpdateActions_WithNewEnumAndOldAsNonEnum_ShouldBuildRemoveAndAddActions() {
+        // preparation
+        final AttributeDefinitionDraft newDefinition = AttributeDefinitionDraftBuilder
+            .of(EnumAttributeType.of(EnumValue.of("foo", "bar")), "a", ofEnglish("new_label"), true)
+            .build();
+        final ProductTypeDraft productTypeDraft = ProductTypeDraftBuilder
+            .of("foo", "name", "desc", singletonList(newDefinition))
+            .build();
+
+        final AttributeDefinition oldDefinition = AttributeDefinitionBuilder
+            .of("a", ofEnglish("new_label"), StringAttributeType.of())
+            .build();
+        final ProductType productType = mock(ProductType.class);
+        when(productType.getAttributes()).thenReturn(singletonList(oldDefinition));
+
+        // test
+        final List<UpdateAction<ProductType>> updateActions = buildAttributesUpdateActions(productType,
+            productTypeDraft, SYNC_OPTIONS);
+
+        // assertions
+        assertThat(updateActions).containsExactly(
+            RemoveAttributeDefinition.of("a"),
+            AddAttributeDefinition.of(AttributeDefinitionDraftBuilder
+                .of(newDefinition.getAttributeType(),
+                    newDefinition.getName(), newDefinition.getLabel(),
+                    newDefinition.isRequired())
+                .build())
+        );
+    }
+
+    @Test
+    void buildAttributesUpdateActions_WithOldLenumAndNewAsNonLenum_ShouldBuildRemoveAndAddActions() {
```

Comment:
What's the difference between this test and the other test `buildAttributesUpdateActions_WithNewLenumAndOldAsNonLenum_ShouldBuildRemoveAndAddActions` that you have to make separate tests?

Reply:
This is the difference: `WithOldLenumAndNewAsNonLenum` and `WithNewLenumAndOldAsNonLenum` :)

It's trivial but they are 2 different cases.

Reply:
But why you test these cases and for example not case like `oldSetAndNewAsLenum` etc.? 😄

Reply:
but this is tested here: `buildAttributesUpdateActions_WithSetOfText_ShouldBuildActions` (implicitly)